### ISSUE TITLE
Dependabot::DependencyFileNotParseable on valid github action workflow files

### ISSUE THREAD

**Issue by uli-noorhakim** [dependabot/dependabot-core#6109]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

github-actions

### Package manager version

-

### Language version

-

### Manifest location and content before the Dependabot update

```yaml
name: Build Gradle Cache

on: 
  workflow_dispatch:
  schedule:
    - cron: '0 0 * * 0' # every Sunday at 00:00

jobs:
  build-gradle-cache:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Building Cache
        uses: gradle/gradle-build-action@v2
        with:
          cache-read-only: false
          arguments: :app:assembleDebug
```

### dependabot.yml content

```yaml
version: 2
updates:
  - package-ecosystem: "github-actions"
    directory: "/"
    reviewers:
      - "Bhinneka/android-reviewers"
    schedule:
      interval: "weekly"
      day: "sunday"

  - package-ecosystem: "gradle"
    directory: "/"
    reviewers:
      - "Bhinneka/android-reviewers"
    schedule:
      interval: "weekly"
      day: "sunday"
```

### What you expected to see, versus what you actually saw

Check is successful, instead I got an generic parsing error

```
Dependabot encountered the following error:

Dependabot::DependencyFileNotParseable
```

The log isn't exactly helpful either

```
  proxy | time="2022-11-11T11:37:13Z" level=info msg="proxy starting" commit=fa80f718f24b195d66d5537781a77f0a8e8cee62
  proxy | 2022/11/11 11:37:13 Listening (:1080)
updater | 2022-11-11T11:37:13.324318669 [508745584:main:WARN:src/devices/src/legacy/serial.rs:214] Detached the serial input due to peer close/error.
updater | time="2022-11-11T11:37:15Z" level=info msg="guest starting" commit=42441b355753c42573b16dc1d5c0da5de9bb45b3
updater | time="2022-11-11T11:37:15Z" level=info msg="starting job..." fetcher_timeout=5m0s job_id=508745584 updater_timeout=45m0s updater_version=fd4f9f6e6fa9e7f8363c7c3d6c84027a9156c3bf
updater | I, [2022-11-11T11:37:16.583562 #8]  INFO -- sentry: ** [Raven] Raven 3.1.2 ready to catch errors
updater | To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
updater | INFO <job_508745584> Starting job processing
  proxy | 2022/11/11 11:37:19 [002] GET https://github.com:443/Bhinneka/aronawa-android/info/refs?service=git-upload-pack
  proxy | 2022/11/11 11:37:19 [002] * authenticating git server request (host: github.com)
  proxy | 2022/11/11 11:37:22 [002] 200 https://github.com:443/Bhinneka/aronawa-android/info/refs?service=git-upload-pack
  proxy | 2022/11/11 11:37:22 [004] POST https://github.com:443/Bhinneka/aronawa-android/git-upload-pack
  proxy | 2022/11/11 11:37:22 [004] * authenticating git server request (host: github.com)
  proxy | 2022/11/11 11:37:22 [004] 200 https://github.com:443/Bhinneka/aronawa-android/git-upload-pack
  proxy | 2022/11/11 11:37:22 [006] POST https://github.com:443/Bhinneka/aronawa-android/git-upload-pack
  proxy | 2022/11/11 11:37:22 [006] * authenticating git server request (host: github.com)
  proxy | 2022/11/11 11:37:22 [006] 200 https://github.com:443/Bhinneka/aronawa-android/git-upload-pack
updater | INFO <job_508745584> Finished job processing
updater | time="2022-11-11T11:37:22Z" level=info msg="task complete" container_id=job-508745584-file-fetcher exit_code=0 job_id=508745584 step=fetcher
updater | I, [2022-11-11T11:37:23.767955 #7]  INFO -- sentry: ** [Raven] Raven 3.1.2 ready to catch errors
updater | To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
updater | INFO <job_508745584> Starting job processing
updater | INFO <job_508745584> Starting update job for Bhinneka/aronawa-android
updater | INFO <job_508745584> Finished job processing
updater | INFO Results:
updater | Dependabot encountered '1' error(s) during execution, please check the logs for more details.
updater | time="2022-11-11T11:37:43Z" level=info msg="task complete" container_id=job-508745584-updater exit_code=0 job_id=508745584 step=updater
```

I have two `yml` files in my workflows folder, I tried removing one, the other still generate the same error. So I guess this happen to all my workflow files.

**Comment by uli-noorhakim** [comment]
Hmm just found #5453 , I'll be trying it out, and close this issue if it works.

**Comment by uli-noorhakim** [comment]
fixed using workaround from #5453, basically put `"` around anything with `spaces` and/or have special character like `:` or `*`, even if it's a valid YAML without `"`.

**Comment by jeffwidman** [comment]
Thanks for circling back to let us know!
